File: language_benchmarks/lcg32.js
if (process.argv.length !== 4) process.exit(2);
const iterations = Number.parseInt(process.argv[2], 10) >>> 0;
let state = Number.parseInt(process.argv[3], 10) >>> 0;
for (let index = 0; index < iterations; index += 1) {
  state = (Math.imul(state, 1664525) + 1013904223) >>> 0;
}
console.log(`result=${state}`);

File: language_benchmarks/lcg32.c
#include <inttypes.h>
#include <stdint.h>
#include <stdio.h>
#include <stdlib.h>

int main(int argc, char **argv) {
    if (argc != 3) return 2;
    const uint32_t iterations = (uint32_t)strtoul(argv[1], NULL, 10);
    uint32_t state = (uint32_t)strtoul(argv[2], NULL, 10);
    for (uint32_t index = 0; index < iterations; ++index) {
        state = state * 1664525u + 1013904223u;
    }
    printf("result=%" PRIu32 "\n", state);
    return 0;
}
